brave-core/src/main/java/com/github/kristofa/brave/LocalTracer.java

```diff
@@ -203,7 +204,7 @@ public void finishSpan(long duration) {
     private void internalFinishSpan(Span span, long duration) {
         synchronized (span) {
             span.setDuration(duration);
-            spanCollector().collect(span);
+            reporter().report(span.toZipkin());
```

Comment:
Is it worth moving the reporting out of the synchronized block (though I wouldn't expect convention on that span)?